kernel/src/syscall.rs

```diff
@@ -132,3 +132,57 @@ pub trait UserspaceKernelBoundary {
         writer: &mut Write,
     );
 }
+
+/// A dummy `UserspaceKernelBoundary` implementation, used to satisfy the type
+/// system when bootstrapping a new arch/chip crate.
+impl UserspaceKernelBoundary for () {
```

Comment:
The cortex-m `UserspaceKernelBoundary` should just work, shouldn't it?

I think adding a new arch is so rare that this wouldn't get used in practice, and anything that needs it doesn't really support Tock enough to be merged into mainline.

Reply:
That seems fine to me? If it is a new chip with a supported architecture, then it makes sense to use it (and if there are no processes on the board the code won't get called anyway). If it is a new chip with an unsupported arch then creating an empty `UserspaceKernelBoundary` implementation is useful because it will need to get filled in eventually anyway to be merged into Tock.